tests/test_utils.py

```diff
@@ -74,6 +75,21 @@ def test_expand_grid_dict(self):
             ],
         )
 
+    def test_convert_to_vary(self):
+        """Test of the convert_to_zip function."""
+        varied = convert_to_vary({"a": [1, 2], "b": [{"c": 3}, {"c": 4}]})
+        self.assertNotEqual(id(varied[0]["b"]), id(varied[2]["b"]))
```

Comment:
This only checks the basic sanity and not the issue where if you change one you dont change the other, right? I wonder if I would change this to "
```
btest = dict(c=4)
varied = convert_to_vary({"a": [1, 2], "b": [{"c": 3}, btest]})
btest["c"] = 5
self assertEqual(varied[1]["b"], dict(c=4))
```

?

Reply:
`self.assertNotEqual(id(` is checking the address of dictionaries, not just a sanity check.

Reply:
you're right, thanks, @dachengx